### ISSUE TITLE
[Translation] Removing the effects of AWC for BLA

### ISSUE THREAD

**Issue by KipCrossing** [soiltechproject/soiltech-tools#6]
**tl;dr?** Easiest way to put it is that we are doing the same steps for BLA YLF; but instead of using the [yield] on the y-axis, we are using the [AWC yield gap] on the y-axis.

**Problem:**  

The common method of performing [Boundary Line Analysis](https://www.cambridge.org/core/journals/journal-of-agricultural-science/article/boundaryline-analysis-of-fieldscale-yield-response-to-soil-properties/5C2C74CA0C6D193CFB0AD8BDA0FD4B34) (BLA) to identify the Yield Limiting Factor (YLF) includes the use of all the potential/available soil attributes that may be yield limiting. 

In most of Australia's dryland cropping regions, the Available Water Capacity (AWC), epically during a dry season, is generally the primary YLF. This becomes a problem as the soil cannot be managed for AWC. If the AWC is the YLF for the majority of the paddock, then land managers are limited in therms of soil management practices/decisions. For example the addition of lime or nitrogen to the soil. 

**Solution:** 

*Note: this thread is for solution discussions and is a WIP; tweaks or differences the the solution may be displayed below.*

To overcome this, we first analyse the yield gap (or yield potential) of the AWC at each point. The yield gap is the difference between the maximum potential yield as per the AWC boundary line and the measured yield at a particular location. Once the yield gap is obtained, we then theorise that the gap may be due to the effects of other soil properties or nutrients. We then perform a new BLA for each soil attribute on the AWC yield gap. Finally, we create zones (patchwork quilt) of the AWC Yield gap Limiting Factors (YgLF) by taking the soil attribute that produced the smallest [AWC yield gap] potential at each location.

**Comment by KipCrossing** [comment]
**Preliminary sketch of the AWC yield gap BLA**

![IMG_4207](https://user-images.githubusercontent.com/7298449/88743871-28980e80-d189-11ea-9b6d-68f49d816521.jpeg)

**Comment by KipCrossing** [comment]
**Additional thoughts:** 

I think we may need to consider that large AWC yield gap potentials at certain locations for a YgLF may be due to unknown/unmeasured soil attributes. For example if the Bulk density was not measured and would of been the AWC YgLF, at that location, another attribute would be selected as the YgLF. Further, the larger the yield gap potential, the more uncertain it is that that attribute is the YgLF. 

Therefore, we may need to put a threshold on the yield gap potential and call values larger than that threshold an unknown YgLF.

**Comment by KipCrossing** [comment]
After having a discussion with @correllink (Shaz) a question was raised:

Should the yield gap (yield deficit in the above sketch) be represented as **negative** values *OR* **positive** values (the distance from the line to the point)? ie.  (1,-3), (2,-6), (3,-9)... *OR* (1,3), (2,6), (3,9)...

If they were **negative**; the boundary line, for the nutrient vs the AWC yield gap, would be situated around the smaller gaps.

If they were **positive**; the boundary line would be situated around the larger gaps.
